backend/s3/config.go

```diff
@@ -73,9 +48,25 @@ func ParseConfig(s string) (interface{}, error) {
 		if url.Scheme == "http" {
 			cfg.UseHTTP = true
 		}
-
-		cfg.Bucket = url.Path[1:]
+		path = strings.SplitN(url.Path[1:], "/", 2)
+	} else if strings.HasPrefix(s, "s3:") {
+	       s = s[3:]
+	       path = strings.SplitN(s, "/",3)
+	       cfg.Endpoint = path[0]
+	       path = path[1:]
+	} else {
+		return nil, errors.New("s3: invalid format")
+	}
+	if len(path) < 1 {
+		return nil, errors.New("s3: invalid format, host/region or bucket name not found")
 	}
+	cfg.Bucket = path[0];
+	if len(path) > 1 {
+	   cfg.Prefix = path[1]
+	} else {
+	   cfg.Prefix = defaultPrefix
+        }
 
 	return cfg, nil
 }
+
```

Comment:
This file is not properly formatted with `gofmt`, please run it on the file and force-push a new commit.